eclass/unpacker.eclass

```diff
@@ -610,7 +615,13 @@ unpacker_src_uri_depends() {
 		*.rar)
 			deps[rar]="app-arch/unrar" ;;
 		*.7z)
-			deps[7z]="app-arch/p7zip" ;;
+			deps[7z]="
+				|| (
+					app-arch/p7zip
+					app-arch/7zip
```

Comment:
fwiw I'd list the better dependency first (7zip), it'll be what portage picks by default